web/modules/custom/farm_nfa/src/BreadcrumbBuilder.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+namespace Drupal\farm_nfa;
+
+use Drupal\Core\Breadcrumb\Breadcrumb;
+use Drupal\Core\Breadcrumb\BreadcrumbBuilderInterface;
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Link;
+use Drupal\Core\Routing\RouteMatchInterface;
+use Drupal\plan\Entity\PlanInterface;
+
+/**
+ * Breadcrumb builder class for generating breadcrumbs.
+ *
+ * Builds breadcrumbs within the context of the abv_app Drupal package.
+ *
+ * @package Drupal\abv_app
+ */
+class BreadcrumbBuilder implements BreadcrumbBuilderInterface {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function applies(RouteMatchInterface $route_match) {
+    $asset = $route_match->getParameter('asset');
+    $plan = $route_match->getParameter('plan');
+    $view = $route_match->getParameter('view_id');
+    if ($asset || $plan || $view == 'farm_asset' || $view == 'farm_plan') {
+      return TRUE;
+    }
+    return FALSE;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function build(RouteMatchInterface $route_match) {
+    $breadcrumb = new Breadcrumb();
+    $links = [];
+    $asset = $route_match->getParameter('asset');
+    $plan = $route_match->getParameter('plan');
+    $breadcrumb->addCacheContexts(['url.path']);
+    $view = $route_match->getParameter('view_id');
+    $links[] = Link::createFromRoute('Home', '<front>');
+    $current_tab = $this->getCurrentTab();
+    if ($asset instanceof EntityInterface || is_numeric($asset)) {
+      $asset = is_numeric($asset) ? \Drupal::entityTypeManager()->getStorage('asset')->load($asset) : $asset;
```

Comment:
Inject the entityTypeManager. See CommentBreadcrumbBuilder.php and comment.services.yml for an example